### PR TITLE
Add in place `add_reaction!` and `add_reactions!` 

### LINKED ISSUE #351: 🚸make `CoreModel` `add_reactions` function more intuitive
Currently `add_reactions` for `CoreModel`s is not like the associate functions in `StandardModel` i.e. vectors and matrices of get added to the structure of `CoreModel`, sort of like a merge. I think it would be better if we fed it an associate map like so:

```julia
add_reactions!(coremodel, ["rxn1" => ["met1"=>stoich1, "met2"=>stoich2,...], ...])
```

So `add_reaction!` and `add_reactions!` will be the same function since the reaction needs to have a name associated with it and the easiest way to do that is with a double map like in the example.

What do y'all think?

### PR DESCRIPTION
**Pull request by stelmo**
Should close #351 when complete. Also brings `CoreModel` more in line with `StandardModel`

### CODE REVIEW

**Comment by cylon-x** [comment]
:red_square: &nbsp;Commit 2648c6e2 requires formatting!

Required formatting changes summary:
```
 src/reconstruction/CoreModel.jl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
```

**Comment by cylon-x** [comment]
:red_square: &nbsp;Commit 1824be05 requires formatting!

Required formatting changes summary:
```
 test/reconstruction/CoreModel.jl | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)
```

**Comment by cylon-x** [comment]
:red_square: &nbsp;Commit 1824be05 requires formatting!

Required formatting changes summary:
```
 test/reconstruction/CoreModel.jl | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)
```

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 118316ca is formatted properly.

File: src/reconstruction/CoreModel.jl [hunk 1]
```diff
@@ -1,3 +1,44 @@
+"""
+    $(FUNCTIONNAME)(model::CoreModel, rxns::Vector{Reaction})
```
Comment (exaexa) [inline src/reconstruction/CoreModel.jl:2]: I guess this could work with $(SIGNATURES)

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 6e5b2811 is formatted properly.

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit ee21b511 is formatted properly.

File: Project.toml [hunk 2]
```diff
@@ -7,6 +7,7 @@ version = "1.0.3"
 Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
 Distributed = "8ba89e20-285c-5b6f-9357-94700520ee1b"
 DistributedData = "f6a0035f-c5ac-4ad0-b410-ad102ced35df"
+DocStringExtensions = "ffbed154-4ef7-542d-bbb7-c09d3a79fcae"
```
Comment (exaexa) [inline Project.toml:10]: was the formatting thing actually fixed?

Comment (exaexa) [reply Project.toml:10]: (if not, let's not add DSEs yet)

File: src/reconstruction/CoreModel.jl [hunk 3]
```diff
@@ -1,3 +1,44 @@
+"""
+    $(TYPEDSIGNATURES)
+
+Add `rxns` to `model` efficiently. Note, the model must already contain the metabolites used by
```
Comment (exaexa) [inline src/reconstruction/CoreModel.jl:4]: remove "Note, " -- the model simply has to contain the metabolites. :D

(same below)

**Review by exaexa — CHANGES_REQUESTED** [review]

File: Project.toml (hunk 2 above)
Comment (stelmo) [reply Project.toml:10]: Formatting not fixed, but the formatting only looks ugly for long docstrings, for shorter ones it looks good.

Comment (stelmo) [reply Project.toml:10]: but you decide, in or out?

Comment (stelmo) [reply Project.toml:10]: I would keep it in, since it is kinda nice

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 4f4e0fd9 is formatted properly.

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit f0d48d3a is formatted properly.

**Pull request merged**